app/main/translate.py

```diff
@@ -7,6 +7,25 @@
 # TODO get rid of these
 from flask import current_app, session
 
+# for Marian, by Dominik:
+from websocket import create_connection
+
+def translate_with_marian_model(model, sentences):
+
+    ws = create_connection("ws://{}/translate".format(model.server))
+
+    batch = ""
+    for sent in sentences:
+        batch += sent + "\n"
```

Comment:
@Gldkslfmsd Marian ma jednu vetu na radek, ta ma nejakou maximalni povolenou delku, ale kolik dostane vet celkem je mu jedno? Ja jen ze se to rozdeli na vety a pak se to zase nejak slepi, tak jestli to necemu nahodou nevadi.

Reply:
jo, je jedno, kolik vět dostane, on si to interně nabatchuje a všecko potom vrátí. 
Mělo by smysl mu posílat menší batche, aby po zpracování první části se to mohlo rovnou vracet klientovi, ale to se tady stejně neděje, tak je to jedno.
Rozděluju to na "věty" tou chytrou funkcí tam dole, aby delší věty než limit (cca 433 znaků) přeložil navíckrát, protože najednou by to mohlo bejt moc pomalý a nepřesný.

Reply:
@Gldkslfmsd ses si tim jistej? Kdyz tomu dam velkej vstup (100kB), tak to sezere nejakejch 5GB pameti. To mi nejak nesedi s tim, ze by si to interne nejak rozsekal a pocital po castech. Hlavne ale, kdyz stejne velkej vstup prijde i na ten druhej model tak uz se na tu kartu nevejdou a jeden z nich umre na oom.
Jak jsi teda dospel k tomu, ze to interne nabatchuje? Mozna to prece jen pouzivalo nektere promenne z `env.sh`?

Reply:
> @Gldkslfmsd ses si tim jistej? Kdyz tomu dam velkej vstup (100kB), tak to sezere nejakejch 5GB pameti. 

Z RAM nebo z GPU?

> To mi nejak nesedi s tim, ze by si to interne nejak rozsekal a pocital po castech. Hlavne ale, kdyz stejne velkej vstup prijde i na ten druhej model tak uz se na tu kartu nevejdou a jeden z nich umre na oom.
Jak jsi teda dospel k tomu, ze to interne nabatchuje? 

marian-server na to má parametry mini-batch a maxi-batch, předpokládal jsem, že fungujou.

> Mozna to prece jen pouzivalo nektere promenne z env.sh?

To určitě ne.